### PR TITLE
75 open api documentation

### PR DESCRIPTION
**Pull request by Schereo**

### CODE REVIEW

File: api/src/test/mocks/identities.ts [hunk 1]
```diff
@@ -192,7 +193,7 @@ export const ServerIdentityMock: IdentityJsonUpdate & { userData: User } = {
 		type: 'ed25519',
 		public: '5r7cbQkEXi2srrNUrVDkB79NnPuiBguWsPDvS6nY7yEb',
 		secret: '6rK7CLKdDw9kBYLQhH4A11vpeS1Hw9jvZagrqgtGcGEp',
-		encoding: 'base58'
+		encoding: Encoding.base16
```
Comment (dominic22) [inline api/src/test/mocks/identities.ts:196]: please also use base58

File: api/tsconfig.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
 		"noUnusedLocals": true,
 		"typeRoots": ["node_modules/@types", "types"],
 		"outDir": "dist",
+		"skipLibCheck": true,
```
Comment (dominic22) [inline api/tsconfig.json:12]: why is that necessary?

File: api/src/tools/open-api-creator/index.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
+// import { getJsonSchemaReader, getOpenApiWriter, makeConverter } from 'typeconv'
```
Comment (dominic22) [inline api/src/tools/open-api-creator/index.ts:1]: please remove obsolete cod

File: api/package.json [hunk 4]
```diff
@@ -39,6 +40,7 @@
   "dependencies": {
     "@iota/identity-wasm": "0.3.4",
     "@sinclair/typebox": "^0.12.8",
+    "@types/swagger-jsdoc": "^6.0.0",
```
Comment (dominic22) [inline api/package.json:43]: this is just a dev dependency like the @types below

File: api/src/index.ts [hunk 5]
```diff
@@ -29,9 +32,78 @@ async function startServer() {
 	const version = CONFIG.apiVersion;
 	const loggerMiddleware = morgan('combined');
 
+	const options = {
```
Comment (dominic22) [inline api/src/index.ts:35]: please move options somewhere else and import them maybe routers or docs is a folder for that

File: api/tsconfig.json (hunk 2 above)
Comment (Schereo) [reply api/tsconfig.json:12]: removed, not necessary anymore

**Comment by dominic22** [comment]
There are some examples which don't have relevant content but I will create another ticket to provide appropriate examples by the swagger docs.

**Comment by dominic22** [comment]
#145 created

**Review by dominic22 — APPROVED** [review]
